core/controllers/learner_dashboard.py

```diff
@@ -146,20 +146,20 @@ class LearnerDashboardFeedbackThreadHandler(base.BaseHandler):
     """Gets all the messages in a thread."""
 
     @acl_decorators.can_access_learner_dashboard
-    def get(self, exploration_id, thread_id):
+    def get(self, thread_id):
         """Handles GET requests."""
-        messages = feedback_services.get_messages(
-            exploration_id, thread_id)
+        messages = feedback_services.get_messages(thread_id)
         author_ids = [m.author_id for m in messages]
         authors_settings = user_services.get_users_settings(author_ids)
 
         message_ids = [m.message_id for m in messages]
         feedback_services.update_messages_read_by_the_user(
-            self.user_id, exploration_id, thread_id, message_ids)
+            self.user_id, thread_id, message_ids)
 
         message_summary_list = []
-        suggestion = feedback_services.get_suggestion(exploration_id, thread_id)
+        suggestion = feedback_services.get_suggestion(thread_id)
 
+        exploration_id = thread_id.split('.')[0]
```

Comment:
As a consequence of this shift, I see that several IDs have changed and we would need some migration jobs. How/when are we planning to do that?

Reply:
Actually, none of the stored IDs are changed. The ID for the FeedbackThreadModel is exp_id.some_string. Earlier, in the services, "some_string" was called thread_id, which was incorrect. The edits I have made just change the concept of thread_id to the entire ID of the FeedbackThreadModel. So no migration would be required at the moment.